protocol/src/main/java/net/md_5/bungee/protocol/MinecraftDecoder.java

```diff
@@ -51,4 +99,44 @@ protected void decode(ChannelHandlerContext ctx, ByteBuf in, List<Object> out) t
             }
         }
     }
+
+    @RequiredArgsConstructor
+    private static class EmptyPacketLogger implements GenericFutureListener<Future<? super Void>>
+    {
+        private final InetAddress address;
+
+        @Override
+        public void operationComplete(Future<? super Void> future) throws Exception
+        {
+            System.err.println( "[" + address.getHostAddress() + "] <-> MinecraftDecoder recieved empty packet, disconnected" );
+        }
+    }
+
+    @RequiredArgsConstructor
+    private static class InvalidPacketIdLogger implements GenericFutureListener<Future<? super Void>>
+    {
+        private final InetAddress address;
+        private final int packetId;
+
+        @Override
+        public void operationComplete(Future<? super Void> future) throws Exception
+        {
+            System.err.println( "[" + address.getHostAddress() + "] <-> Varint21FrameDecoder recieved invalid packet id " + packetId + ", disconnected" );
```

Comment:
Varint21FrameDecoder gets packet length, not id?

Reply:
Sure, but you are currently in MinecraftDecoder class

Reply:
... I blame excessive sleepiness!

Reply:
I just noticed my copy-paste error now xd.